ui/src/views/user/survey-edit/SurveyEditView.vue

```diff
@@ -264,6 +295,18 @@ export default {
     onClickSettingsButton() {
       this.isBottomSheetShown = true;
       this.bottomSheetContent = "settings";
+    },
+    getSurveyApi(surveyId) {
+      console.log("running getSurveyApi() with param:");
+      console.log(surveyId);
+      userGetSurvey(surveyId)
+        .then((response) => {
+          console.log(JSON.stringify(response));
+          // how to load the response into the formData?
```

Comment:
We can load the survey data in the view, then set survey as props in form builder and settings. Take into account the mobile view as well. Make sure that the component gets rerendered on Props change.

Please tell me if you need assistance in this XD

Reply:
I think I get the concept, but I still need to wait for the 'official' getSurvey and editSurvey apis from BE so that I can try and test. For this PR I won't include the edit survey functionality yet, only create survey.

Reply:
If we merged https://github.com/haanh764/surveyapp/pull/48, we can make use of api mock function XD no need to wait for BE, especially since Bartek already gave his expected result XD  but we'll see.